test/util.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package test
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+func AreEqualJSON(s1, s2 string) (bool, error) {
```

Comment:
I guess  the philosophical quesiton here is - do you want to include the formatting as something to be explicitly tested?  Is, for example, the number of spaces of indent on JSON files part of the spec?  Historically I've tried to just compare the data, but I'm happy to go whichever way you want.  (I was, by the way, getting all errors when comparing to the golden files.  Maybe an extra newline or soemthing.  I didn't dig as I was adding this anyway.)

I'll leave it in on this next pass, and if you repeat you want it pulled, I'll be happy to do so.